### PR TITLE
feat: new design of the country leadingboard #331

### PR DESCRIPTION
**Pull request by arcadioramos**
# Description

[comment]: # The new design was created based on the one from the "New design of the country leading board #331" issue, I've implemented responsive design, also non-hard coded data to display the new countries, just passing the prop to the map function.
Ribbons component was created to make leaderboard code more readable.
Tree_icon.svg was added also to make the code more readable.

Fixes # New design of the country leading board #331

## Type of change

[comment]: # 'Please delete options that are not relevant.'

- [x] New feature (non-breaking change which adds functionality)

### Before
image

### After
1920*1080:
image

Iphone X view : 
image

iPad view: 
image

# How Has This Been Tested?

- [x] Cypress component tests

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by VWRoli — CHANGES_REQUESTED** [review]
~~You changed the filter component in this PR which you did in another PR. When you create a new branch for a new issue go back to the main branch and start the new feature branch from there. For now, just copy the old filter component code into this one so you don't change the part of the code that has nothing to do with this issue.~~
EDIT: I just saw that it contains a different PR. Don't worry about this, my bad.

https://github.com/Greenstand/treetracker-web-map-client/blob/583b005ec650687cd9db2f9b54367a380aa5ef87/src/components/LeaderBoard.js#L77-L85
Typography is mostly responsive on its own. No need to do this. Just give it a variant "H5", and it will be good. Check the [theme](https://github.com/Greenstand/treetracker-web-map-client#design-sandbox) and the [Figma](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap?node-id=2497%3A9322) design for reference.

___

Add a margin between the table header and table content.

___

I see the nonTop3 is one component, but the top 3 is three separate components that basically do the same thing as the nonTop3. Create in this file (as we will use it only here) a component that can handle the top 3 and the rest of it. Only the icon is changing.
Remember if you have to copy/paste code you are probably making a mistake. DRY principle...

___

Take a good look at the design file, both the mobile and the desktop versions. Typography colors, sizes, etc... 
Also on the mobile, you should do something with the grid system. It doesn't look good on my side.
Screenshot from 2021-12-20 14-29-39

**Comment by arcadioramos** [comment]
Thank you @VWRoli , I'll try to solve these as soon as possible 😄  !

**Comment by arcadioramos** [comment]
I've worked on the comments you gave me @VWRoli !

**Comment by VWRoli** [comment]
I was trying to lead you in the direction to forget the `customText` class altogether not to add another one:D There is no Montserrat font family that you have to add at all. Also, there is no need to center the country name, it wasn't centered before, was it?

Align the `Country` tag so it's above the flag. Actually, even though the trophies are different you can still position them in the same way with their rank. Create a box for the icon and the rank and position them according to the design.

The flag border-radius is 8px on the big screens and 4px on small screens. Follow the [Figma](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap?node-id=2497%3A9322) design.
 
Margins between the leaderboard items are also too small. And they are different based on screen sizes. You can find all the resources in the  [Figma](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap?node-id=2497%3A9322) design.

The tree icons should be closer to the size of the text, also align them. 

We don't have borders on the items, just the shadow. Remove these, please.
https://github.com/Greenstand/treetracker-web-map-client/blob/4abbad049bfb1470d1427716022835f4bf12bbce/src/components/LeaderBoard.js#L61
https://github.com/Greenstand/treetracker-web-map-client/blob/4abbad049bfb1470d1427716022835f4bf12bbce/src/components/LeaderBoard.js#L103
https://github.com/Greenstand/treetracker-web-map-client/blob/4abbad049bfb1470d1427716022835f4bf12bbce/src/components/LeaderBoard.js#L99

Basically, my best advice is to follow the [Figma](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap?node-id=2497%3A9322) design.

**Comment by arcadioramos** [comment]
Thank you for the advice, I'm currently working on the comments!

**Comment by arcadioramos** [comment]
Should I commit my stuff with this way? I'm looking for approval before commit and push to my branch.
Iphone X view : 
image
1920*1080:
image
Ipad: 
image

**Comment by VWRoli** [comment]
The number 4 is not aligned with the rest of the line. The badges should be absolutely positioned to the top left corner, and be a little bit bigger.
Bottom (or top) margin is missing: (use `theme.spacing()` here!)
Screenshot from 2021-12-21 15-43-10

The items should also have a top/bottom margin.
The tree icon is still a bit bigger compared to the text. Should have some margin from the text too.
The box shadow of the item is `box-shadow: 0px 4px 10px 0px #00000026;` In figma you can check out the CSS for help.

The leaderboard item can be a component on its own which accepts a couple of props, in the same file, so it's easier to read.

But you are definitely getting closer!

**Comment by arcadioramos** [comment]
- The number 4 is not aligned with the rest of the line.  -Done
- The badges should be absolutely positioned to the top left corner, and be a little bit bigger. - Don't know how to do this without destroying all the alignment all the other stuff has.
- Bottom (or top) margin is missing: (use theme.spacing() here!) - Done but with margin "sx={{my:6}}"
- The items should also have a top/bottom margin. - Done
- The tree icon is still a bit bigger compared to the text. Should have some margin from the text too. - Done
- The box shadow of the item is box-shadow: 0px 4px 10px 0px #00000026. -Done

Here's what I have by this point: 
1920*1080: 
image
IPAD: 
image
Iphone X: 
image

What do you recommend to me to make it better to be able to commit and merge this?

**Comment by VWRoli** [comment]
Is the borderRadius of the flags on small screens are 4px?
What I would try with the badges and numbers (4 is still misplaced by the way)  is to create a container element for the number. That container either has the number centered:
Screenshot from 2021-12-21 18-04-17

Or a placeholder element, actually if we have a number container we don't need a placeholder. The container is the placeholder itself. So if we don't have a number we can position the badge to the top left.
Screenshot from 2021-12-21 18-08-09

Margins and padding are not random numbers by the way. Follow the Figma design where you can read the margins and paddings. 

> Done but with margin "sx={{my:6}}"

This should be `56px` on big screen and `28px` on small screen.

Also for some reason there is a white line under the flags which should not be there.

**Comment by VWRoli** [comment]
Is the borderRadius of the flags on small screens are 4px?
What I would try with the badges and numbers (4 is still misplaced by the way)  is to create a container element for the number. That container either has the number centered:
Screenshot from 2021-12-21 18-04-17

Or a placeholder element, actually if we have a number container we don't need a placeholder. The container is the placeholder itself. So if we don't have a number we can position the badge to the top left.
Screenshot from 2021-12-21 18-08-09

Margins and padding are not random numbers by the way. Follow the Figma design where you can read the margins and paddings. 

> Done but with margin "sx={{my:6}}"

This should be `56px` on big screen and `28px` on small screen.

Also for some reason there is a white line under the flags which should not be there.

**Comment by arcadioramos** [comment]
Hello my Hungarian fella, here is what I've done.
I followed the Figma mobile and desktop designs, it was better and my bad that I've never seen the "mobile" window before and I was trying to guess the sizes.
I centered the numbers as you mentioned, and I've put the badges top left corner as you mentioned too.
I removed the white border on the flag as you mentioned.
I added the margins as you mentioned and as the Figma design was.
The border-radius on flags changes from 8px to 4px as you wanted.

I hope you like what you see, and if not I'm open to comments on changing stuff.
If everything is okay let me know to commit my changes and make the pull request, thank you mate !.

iPhone X: 
image
1920*1080: 
image
Ipad: 
image

File: src/components/LeaderBoard.js [hunk 1]
```diff
@@ -163,139 +53,120 @@ function LeaderBoard(props) {
...
+                </Typography>
+              </Grid>
+            </Grid>
           </Box>
-        ))}
-      <br></br>
-    </Grid>
+        ))
+      ) : (
+        <p>Siuuu</p>
```
Comment (QuaidBartolomei) [inline src/components/LeaderBoard.js:167]: Is this for error handling? If so it should have a more appropriate message so that the purpose is clear to other developers. If this is only here to allow the use of the ternary operator then it should be switched back to the [neater form of short circuiting.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND)

File: src/components/common/Filter.js [hunk 2]
```diff
@@ -260,6 +268,13 @@ function Filter(props) {
           </Box>
         </Box>
       </form>
+      {showMessage ? (
+        <Alert severity="warning" sx={{ my: 4 }}>
+          Start date: {startDate} is bigger than end date: {endDate}
+        </Alert>
+      ) : (
+        <p></p>
+      )}
```
Comment (QuaidBartolomei) [inline src/components/common/Filter.js:277]: another place for short-circuiting

```jsx
     {showMessage && (
        <Alert severity="warning" sx={{ my: 4 }}>
          Start date: {startDate} is bigger than end date: {endDate}
        </Alert>
      )}
```

File: src/components/LeaderBoard.js [hunk 3]
```diff
@@ -163,139 +53,120 @@ function LeaderBoard(props) {
...
-            onClick={() => handleCountryClick(countries[1].id)}
-            className={`${classes.rectangle72}  ${classes.clickable}`}
+            sx={{
+              flexGrow: 1,
+              borderRadius: 16,
+              borderColor: 'grey.500',
+              boxShadow: 5,
+            }}
+            margin={2}
+            borderColor={'beige.500'}
```
Comment (QuaidBartolomei) [inline src/components/LeaderBoard.js:103]: `borderColor` is defined twice here. Because this component already has the `sx` prop I think it is better to be consistent and keep all styles for this component inside the sx object. It's also important to be aware of the unique differences between the sx prop and makeStyles. You can read about those here https://mui.com/system/the-sx-prop/ specifically be aware of the borderRadius difference 

```
The borderRadius properties multiples the value it receives by the theme.shape.borderRadius value (the default for the value is 4px).
```

File: src/components/LeaderBoard.js [hunk 4]
```diff
@@ -163,139 +53,120 @@ function LeaderBoard(props) {
...
-            <Box className={classes.planter}>Planter</Box>
-            <Box className={classes.content}>{countries[2].name}</Box>
-            <Box className={classes.rating}>
-              Planted
-              <br></br>
-              {countries[2].planted}
-            </Box>
-          </Box>
-        </Box>
+          <Grid item xs={1}>
```
Comment (QuaidBartolomei) [inline src/components/LeaderBoard.js:71]: Instead of using an empty grid column for spacing, add margins to the Grid's container (in this case the Box on line 57)

**Review by QuaidBartolomei — CHANGES_REQUESTED** [review]
@arcadioramos Thanks for helping us with improving this component. There are still some layout issued present in your screenshots. It is important to try and match the layout of the design document as close as possible. When using Figma make sure you are signed in so that you can inspect the properties of each design element, as well as the spacing between each element. 

Looking at the mobile screenshots the gold/silver/bronze ribbons are too small and not positioned correctly (you will probably need to use absolute positioning or negative margin values for these).

There is also too much space between the country list and the page/screen border in the mobile screenshot. Compare closely to the design doc.

File: src/components/LeaderBoard.js (hunk 1 above)
Comment (arcadioramos) [reply src/components/LeaderBoard.js:167]: This is not the latest version of my code, that part of the code doesn't exist anymore, I haven't pushed my changes to the remote because I don't wanna fill the pull request with commits, but if you want me to push the changes to see them closely I'll do it.

File: src/components/common/Filter.js (hunk 2 above)
Comment (arcadioramos) [reply src/components/common/Filter.js:277]: I did this in the newest changes, haha !

File: src/components/LeaderBoard.js (hunk 4 above)
Comment (arcadioramos) [reply src/components/LeaderBoard.js:71]: Thank you

File: src/components/LeaderBoard.js (hunk 1 above)
Comment (arcadioramos) [reply src/components/LeaderBoard.js:167]: The problem with the ribbons is that I copied as an SVG directly from Figma and if I move them they tend to disappear, or I don't know if it is because of my inexperience.

Comment (arcadioramos) [reply src/components/LeaderBoard.js]: Could you look at my last commit @QuaidBartolomei? I think I need some help, I'm kinda blocked trying to make this as 1:1 as possible and also trying to make it responsive :/

**Comment by VWRoli** [comment]
I don't know how exactly, but I'm sure it is possible to extract the SVG's into their own file into the images folder. That would be great and would make the file readability much better.

`fontFamily: 'Lato'` is not necessary anywhere, it already is Lato. You can delete them all.

UPDATE: It's actually not working but that is a theme issue, not yours. You don't have to set font family at all.

You don't have to set font sizes. H5 is already 20px and 16px on mobile, check out the theme.js. So basically the whole `text` class becomes redundant.

Position the `#` sign correctly!

File: src/components/LeaderBoard.js (hunk 1 above)
Comment (VWRoli) [reply src/components/LeaderBoard.js]: > The problem with the ribbons is that I copied as an SVG directly from Figma and if I move them they tend to disappear, or I don't know if it is because of my inexperience.

Just give them a class and position them absolutely with a negative margin and proper size.

**Comment by arcadioramos** [comment]
Hello comrades, I've worked on what you told me to change.

- Ribbons bigger en aligned top-left. done
- space between the country list and the page/screen border in the mobile screenshot. done
- Font-family removed to not be redundant. done
- Position the # correctly. done

1920*1080: 
image
Ipad: 
image
iPhone X: 
image

This is how the component will be looking once you solve the theme issue.

**Comment by arcadioramos** [comment]
Hello comrades, I've worked on what you told me to change.

- Ribbons bigger en aligned top-left. done
- space between the country list and the page/screen border in the mobile screenshot. done
- Font-family removed to not be redundant. done
- Position the # correctly. done

1920*1080: 
image
iPhone X: 
image
Ipad: 
image

**Comment by dadiorchen** [comment]
I just want to say, CSS is not so easy...

**Comment by dadiorchen** [comment]
And thank you for all your guy's effort on this, and also, I believe that when our new web map lunched to prod, we all will be proud of it! It would be a beautiful work, and most importantly it's an output of a dev community from all around the world.

**Comment by VWRoli** [comment]
The negative positioning should be towards the top, not the left.

**Comment by arcadioramos** [comment]
I've moved it to the top as you asked for :) 
Anything else sir? 

1920*1080: 
image
iPad: 
image
iPhone X: 
image

**Comment by arcadioramos** [comment]
I've moved it to the top as you asked :).
Anything else sir? 

1920*1080:
image
iPad: 
image
iPhone X: 
image

**Comment by dadiorchen** [comment]
@VWRoli when you review PR, will you download the code and run it?

**Comment by dadiorchen** [comment]
And I noticed that the design changed when we working on it, I will report this to our designer and find a way to avoid this.

**Comment by dadiorchen** [comment]
Our another page, try to find the difference: 
image

image

**Comment by VWRoli** [comment]
> @VWRoli when you review PR, will you download the code and run it?

Of course.

**Comment by VWRoli** [comment]
> Our another page, try to find the difference: image
> 
> image

I noticed this when doing the CSS update of the project. It is on my to-do list.

**Comment by dadiorchen** [comment]
> > @VWRoli when you review PR, will you download the code and run it?
> 
> Of course.

I'm trying to automate this to generate some screenshot or video when new PR coming in.

**Comment by VWRoli** [comment]
> Bottom (or top) margin is missing: (use theme.spacing() here!)

This is still missing, nowhere do you use `theme.spacing`. Margins, padding, and positioning should be done with `theme.spacing`

___

The ribbons are still smaller than they should be. Both on big and small screens. 
We can create a reusable component for the ribbon that accepts a fill prop for the color and returns an SVG. `<Ribbon fill="#FFD700" />`. This way we don't have to write the SVG into the JSX, which is a bad practice IMO.

Ribbon.js
```
import * as React from 'react';
import { makeStyles } from 'models/makeStyles';

const useStyles = makeStyles()((theme) => ({
  root: {
    [theme.breakpoints.down('md')]: {
      width: 32,
      height: 40,
    },
  },
}));

function Ribbon(props) {
  const { fill } = props;
  const { classes } = useStyles();
  return (
    <svg
      width="63"
      height="80"
      viewBox="0 0 63 80"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      className={classes.root}
    >
      <path
        d="M23.8 59.7C24.4 59.9 25.2 60.6 26 61.3L17.4 79.5L11.2 71L0.6 71.5L10 51.7C10.3 51.8 10.5 52 10.7 52.2C11.2 52.7 11.6 53.9 12 55.1C12.6 56.9 13.2 59 15 60C15.7 60.4 16.5 60.6 17.5 60.6C18.6 60.6 19.7 60.3 20.8 60.1C21.7 59.9 22.6 59.7 23.4 59.7C23.4 59.7 23.7 59.7 23.8 59.7ZM52.1 52.1C51.6 52.6 51.2 53.8 50.8 55C50.2 56.8 49.6 58.9 47.8 59.9C47.1 60.3 46.3 60.5 45.3 60.5C44.2 60.5 43.1 60.2 42 60C41.1 59.8 40.2 59.6 39.4 59.6C39.2 59.6 39 59.6 38.8 59.7C38.2 59.9 37.4 60.6 36.6 61.3L45.4 79.5L51.6 71L62.2 71.5L52.8 51.7C52.5 51.8 52.3 52 52.1 52.1ZM58.5 47.1C57.2 49.3 52.7 49.2 50.9 51C49.1 52.8 49.2 57.3 47 58.6C44.8 59.9 41 57.5 38.5 58.2C36.1 58.8 34 62.8 31.4 62.8C28.8 62.8 26.7 58.8 24.2 58.2C21.7 57.5 17.9 59.9 15.7 58.6C13.5 57.3 13.6 52.8 11.8 51C10 49.2 5.5 49.3 4.2 47.1C2.9 44.9 5.3 41.1 4.6 38.6C3.9 36.2 0 34.1 0 31.4C0 28.8 4 26.7 4.6 24.2C5.3 21.7 2.9 17.9 4.2 15.7C5.5 13.5 10 13.6 11.8 11.8C13.6 10 13.5 5.5 15.7 4.2C17.9 2.9 21.7 5.3 24.2 4.6C26.6 3.9 28.7 0 31.4 0C34 0 36.1 4 38.6 4.6C41.1 5.3 44.9 2.9 47.1 4.2C49.3 5.5 49.2 10 51 11.8C52.8 13.6 57.3 13.5 58.6 15.7C59.9 17.9 57.5 21.7 58.2 24.2C58.9 26.6 62.8 28.7 62.8 31.4C62.8 34 58.8 36.1 58.2 38.6C57.4 41.1 59.8 44.9 58.5 47.1ZM52.5 31.5C52.5 19.8 43 10.4 31.4 10.4C19.8 10.4 10.3 19.8 10.3 31.5C10.3 43.2 19.7 52.6 31.4 52.6C43.1 52.6 52.5 43.1 52.5 31.5Z"
        fill={fill}
      />
    </svg>
  );
}

export default Ribbon;

```
Positioning it would happen like so:

```
<div className={classes.ribbons}>
                    <Ribbon fill="#FFD700" />
                    <Typography variant="h5" className={classes.rank}>
                      {index + 1}
                    </Typography>
                  </div>
```

```
ribbons: {
    position: 'relative',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: theme.spacing(-15),
    [theme.breakpoints.down('md')]: {
      marginTop: theme.spacing(-10),
    },
  },
  rank: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    marginTop: theme.spacing(-2),
    [theme.breakpoints.down('md')]: {
      marginTop: theme.spacing(-1),
    },
  },
```

___

https://github.com/Greenstand/treetracker-web-map-client/blob/d0b4b1e9a4c43d15552b6d8bbec02a95d0a9c1da/src/components/LeaderBoard.js#L381

I'm guessing this is just for the bottom padding, right? If so, then we can just add padding-bottom to the last item, or instead of returning a React fragment, we return a Box component that has some bottom padding. Either solution is fine.
`<Box pb={4}>`

___

https://github.com/Greenstand/treetracker-web-map-client/blob/d0b4b1e9a4c43d15552b6d8bbec02a95d0a9c1da/src/components/LeaderBoard.js#L51
Instead of using hard coded values you can use the MUI default breaking points.
`[theme.breakpoints.down('md')]: {
     marginTop: theme.spacing(10),
    },`

___

You can export the tree icon into an SVG file. You can then import it like so:
`import treeIcon from '../images/tree_icon.svg';`
Ans use it like so:
`{'TREES PLANTED '} tree icon`

___

Also, the code got updated so please solve the merge conflict.

**Review by VWRoli — CHANGES_REQUESTED** [review]
> Bottom (or top) margin is missing: (use theme.spacing() here!)

This is still missing, nowhere do you use `theme.spacing`. Margins, padding, and positioning should be done with `theme.spacing`

___

The ribbons are still smaller than they should be. Both on big and small screens. 
We can create a reusable component for the ribbon that accepts a fill prop for the color and returns an SVG. `<Ribbon fill="#FFD700" />`. This way we don't have to write the SVG into the JSX, which is a bad practice IMO.

Ribbon.js
```
import * as React from 'react';
import { makeStyles } from 'models/makeStyles';

const useStyles = makeStyles()((theme) => ({
  root: {
    [theme.breakpoints.down('md')]: {
      width: 32,
      height: 40,
    },
  },
}));

function Ribbon(props) {
  const { fill } = props;
  const { classes } = useStyles();
  return (
    <svg
      width="63"
      height="80"
      viewBox="0 0 63 80"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      className={classes.root}
    >
      <path
        d="M23.8 59.7C24.4 59.9 25.2 60.6 26 61.3L17.4 79.5L11.2 71L0.6 71.5L10 51.7C10.3 51.8 10.5 52 10.7 52.2C11.2 52.7 11.6 53.9 12 55.1C12.6 56.9 13.2 59 15 60C15.7 60.4 16.5 60.6 17.5 60.6C18.6 60.6 19.7 60.3 20.8 60.1C21.7 59.9 22.6 59.7 23.4 59.7C23.4 59.7 23.7 59.7 23.8 59.7ZM52.1 52.1C51.6 52.6 51.2 53.8 50.8 55C50.2 56.8 49.6 58.9 47.8 59.9C47.1 60.3 46.3 60.5 45.3 60.5C44.2 60.5 43.1 60.2 42 60C41.1 59.8 40.2 59.6 39.4 59.6C39.2 59.6 39 59.6 38.8 59.7C38.2 59.9 37.4 60.6 36.6 61.3L45.4 79.5L51.6 71L62.2 71.5L52.8 51.7C52.5 51.8 52.3 52 52.1 52.1ZM58.5 47.1C57.2 49.3 52.7 49.2 50.9 51C49.1 52.8 49.2 57.3 47 58.6C44.8 59.9 41 57.5 38.5 58.2C36.1 58.8 34 62.8 31.4 62.8C28.8 62.8 26.7 58.8 24.2 58.2C21.7 57.5 17.9 59.9 15.7 58.6C13.5 57.3 13.6 52.8 11.8 51C10 49.2 5.5 49.3 4.2 47.1C2.9 44.9 5.3 41.1 4.6 38.6C3.9 36.2 0 34.1 0 31.4C0 28.8 4 26.7 4.6 24.2C5.3 21.7 2.9 17.9 4.2 15.7C5.5 13.5 10 13.6 11.8 11.8C13.6 10 13.5 5.5 15.7 4.2C17.9 2.9 21.7 5.3 24.2 4.6C26.6 3.9 28.7 0 31.4 0C34 0 36.1 4 38.6 4.6C41.1 5.3 44.9 2.9 47.1 4.2C49.3 5.5 49.2 10 51 11.8C52.8 13.6 57.3 13.5 58.6 15.7C59.9 17.9 57.5 21.7 58.2 24.2C58.9 26.6 62.8 28.7 62.8 31.4C62.8 34 58.8 36.1 58.2 38.6C57.4 41.1 59.8 44.9 58.5 47.1ZM52.5 31.5C52.5 19.8 43 10.4 31.4 10.4C19.8 10.4 10.3 19.8 10.3 31.5C10.3 43.2 19.7 52.6 31.4 52.6C43.1 52.6 52.5 43.1 52.5 31.5Z"
        fill={fill}
      />
    </svg>
  );
}

export default Ribbon;

```
Positioning it would happen like so:

```
<div className={classes.ribbons}>
                    <Ribbon fill="#FFD700" />
                    <Typography variant="h5" className={classes.rank}>
                      {index + 1}
                    </Typography>
                  </div>
```

```
ribbons: {
    position: 'relative',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: theme.spacing(-15),
    [theme.breakpoints.down('md')]: {
      marginTop: theme.spacing(-10),
    },
  },
  rank: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    marginTop: theme.spacing(-2),
    [theme.breakpoints.down('md')]: {
      marginTop: theme.spacing(-1),
    },
  },
```

___

https://github.com/Greenstand/treetracker-web-map-client/blob/d0b4b1e9a4c43d15552b6d8bbec02a95d0a9c1da/src/components/LeaderBoard.js#L381

I'm guessing this is just for the bottom padding, right? If so, then we can just add padding-bottom to the last item, or instead of returning a React fragment, we return a Box component that has some bottom padding. Either solution is fine.
`<Box pb={4}>`

___

https://github.com/Greenstand/treetracker-web-map-client/blob/d0b4b1e9a4c43d15552b6d8bbec02a95d0a9c1da/src/components/LeaderBoard.js#L51
Instead of using hard coded values you can use the MUI default breaking points.
`[theme.breakpoints.down('md')]: {
     marginTop: theme.spacing(10),
    },`

___

You can export the tree icon into an SVG file. You can then import it like so:
`import treeIcon from '../images/tree_icon.svg';`
Ans use it like so:
`{'TREES PLANTED '} tree icon`

___

Also, the code got updated so please solve the merge conflict.

**Comment by arcadioramos** [comment]
I updated the pull request description with the new images of the component, if you have any comments I'm open to working on them.

**Comment by arcadioramos** [comment]
I did the last commit because I noticed that I didn't solve the merge conflicts, and with the changes in this commit they should be solved, so I apologize because of not solving it well at the first time but hopefully this time it will be solved and the CI tests will pass :)

**Review by VWRoli — APPROVED** [review]

**Comment by dadiorchen** [comment]
@arcadioramos the last thing, some new conflict emerged, I guess that's because I just merged another PR, please solve it, then I will merge this beautiful PR!

Some hints on this in case you don't know: 
you can merge the web-map-site into your local branch, fix the conflict manually, then just push your new code.

**Comment by arcadioramos** [comment]
Thank you @dadiorchen for your words, I appreciate them it has been a pleasure to contribute to this project :) <3

I already solved the merge conflicts and thank you for the hint!
